app/services/resource_service.rb

```diff
@@ -12,78 +12,126 @@ def initialize(data_provider:)
   def create(resource_class, params)
     @params = params
 
+    # cleanup params for given :id
+    @old_resource_id = @params.delete(:id)
+
     # Wenn die Rolle Restricted eine Information anlegt,
     # so ist diese per default nicht sichtbar, es sei denn
     # das Attribute 'visible' wird in den params mitgegeben und ist 'true'
-    if data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
+    if @data_provider.user.restricted_role? && resource_class.respond_to?(:visible)
```

Comment:
globaler variablenzugriff nötig, wenn `attr_accessor :data_provider`? welcher vorteil? könnte sonst in den private methods (die noch nicht private sind) nicht drauf zugegriffen werden? doch oder?

- [ ] methods als private setzen

Reply:
gilt auch für `resource` und `resource_class`

Reply:
über die globalen Variablenbezeichnungen mit "@" erkennt man meiner Meinung nach viel besser, dass es sich um eine Variable handelt, die in ganzen Serviceklasse verfügbar ist als ausschließlich über die Definition als :attr_accessor.

Der :attr_accessor hat zusätzlich aber noch die Funktion einen einfachen Zugriff dieser Variablen von außerhalb der ServiceKlasse zu geben. Ob das hier allerdings noch benötigt wird, weiß ich nicht. 

Ich vermute in privaten Methoden kann auch über den :attr_accessor zugegriffen werden.